File: C++ solutions/2610. Convert an Array Into a 2D Array With Conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        vector<vector<int>> v;
        map<int, int> m;  
        for (int i = 0; i < nums.size(); i++) m[nums[i]]++;
        int mx=0;
        for(auto k:m) mx=max(mx,k.second);
    
        for(int i=0;i<mx;i++){
            vector<int>vv;
            for(auto k:m){
               if(k.second!=0){
                    vv.push_back(k.first);
                    m[k.first] = k.second - 1;
               }
            }
            if(vv.size()!=0) v.push_back(vv);
        }
        return v;
    }
};

File: JavaScript solutions/2610. Convert an Array Into a 2D Array With Conditions.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var findMatrix = function (nums) {
  const m = new Map();
  nums.forEach((num) => {
    m.set(num, (m.get(num) || 0) + 1);
  });

  let mx = 0;
  for (const count of m.values()) {
    mx = Math.max(mx, count);
  }

  const v = [];
  for (let i = 0; i < mx; i++) {
    const vv = [];
    for (const [key, value] of m.entries()) {
      if (value !== 0) {
        vv.push(key);
        m.set(key, value - 1);
      }
    }
    if (vv.length !== 0) {
      v.push(vv);
    }
  }

  return v;
};
